tsl/src/continuous_aggs/invalidation.c

```diff
@@ -521,6 +521,23 @@ invalidation_entry_set_from_cagg_invalidation(Invalidation *entry, const TupleIn
 						   Form_continuous_aggs_materialization_invalidation_log);
 }
 
+/*
+ * Check if two invalidations can be merged into one.
+ *
+ * Since invalidations are inclusive in both ends, two adjacent invalidations
+ * can be merged.
+ */
+static bool
+invalidations_can_be_merged(const Invalidation *a, const Invalidation *b)
+{
+	/* To account for adjacency, expand one window 1 step in each
+	 * direction. This makes adjacent invalidations overlapping. */
+	int64 a_start = int64_saturating_sub(a->lowest_modified_value, 1);
+	int64 a_end = int64_saturating_add(a->greatest_modified_value, 1);
+
+	return a_end >= b->lowest_modified_value && a_start <= b->greatest_modified_value;
+}
```

Comment:
Is it guaranteed that the invalidations are ordered so that `a << b`? If that is the case, an assertion would be good, if not, testing both orders is probably good.

Reply:
Not sure I understand the question. This function checks for overlap of two ranges and order shouldn't matter. Either they overlap or they don't.